cylc/flow/task_job_mgr.py

```diff
@@ -1087,7 +1097,8 @@ def _prep_submit_task_job(
         Returns:
             * itask - preparation complete.
             * None - preparation in progress.
-            * False - perparation failed.
+            * False - preparation failed.
+            * True - preparation failed because no platforms were found.
```

Comment:
How about returning (return not raise) the `NoPlatformsError` instead of `True`? I think that would be clearer

Reply:
Yes.

Reply:
Argdoc needs updating to reflect this.

Reply:
```suggestion
            * NoPlatformsError - preparation failed because no
              platforms were found.
```